testsuite/lib/libffi.exp

```diff
@@ -225,20 +225,21 @@ proc libffi_target_compile { source dest type options } {
 
 # TEST should be a preprocessor condition.  Returns true if it holds.
 proc libffi_feature_test { test } {
-    set src "ffitest.c"
+    set src "ffitest[pid].c"
 
     set f [open $src "w"]
     puts $f "#include <ffi.h>"
     puts $f $test
-    puts $f "xyzzy"
+    puts $f "/* OK */"
+    puts $f "#else"
+    puts $f "# error Failed $test"
     puts $f "#endif"
     close $f
 
-    set lines [libffi_target_compile $src "" "preprocess" ""]
+    set lines [libffi_target_compile $src /dev/null assembly ""]
```

Comment:
Hi @tromey, thanks for having a look!

Heh, Mike Stump had a very similar question on the GCC list.  I found the `/dev/null` idiom used in several places in the GCC testsuite, see <http://news.gmane.org/find-root.php?message_id=%3C87si0g4kdg.fsf%40hertz.schwinge.homeip.net%3E>.

Using just `""` is fine for proprocess mode (which writes all output to `stdout`), but any other mode would run into `[dejagnu]/target.exp:default_target_compile`:

    proc default_target_compile {source destfile type options} {
        [...]
        if { $destfile == "" && $type != "preprocess" && $type != "none" } {
            error "Must supply an output filename for the compile to default_target_compile"
        }